### ISSUE TITLE
Identifiers beginning with underscores are UB

### ISSUE THREAD

**Issue by fuchsia36** [libtom/libtomcrypt#448]
### Prerequisites

* [x] Checked the developer manual
* [x] Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=repo%3Alibtom%2Flibtomcrypt
* [x] Checked that your issue isn't related to TomsFastMath's limitation that PK operations can by default only be done with max. 2048bit keys

### Description

According to the C specification, **7.1.3 Reserved identifiers**:
>* All identifiers that begin with an underscore and either an uppercase letter or another underscore are always reserved for any use.  
>* All identifiers that begin with an underscore are always reserved for use as identifiers with file scope in both the ordinary and tag name spaces

>If the program declares or defines an identifier in a context in which it is reserved (other than as allowed by 7.1.4), or defines a reserved identifier as a macro name, **the behavior is undefined.**  

According to `grep`, there are around 300 such reserved identifiers used. You can verify this by running `grep -r ' _[A-Za-z_0-9]*' --only-matching --no-filename | sort | uniq`. Some of them are C's or GNU's, but most of them are defined in the project files.

If `libtomcrypt` is to steer clear of UB, they should be removed. I would suggest removing the underscores altogether where possible. In some places it might cause some collisions, and then a quick solution is to just prepend it with any other character, for instance turning `_safer_ecb_decrypt` into `U_safer_ecb_decrypt`.

**Comment by fuchsia36** [comment]
`_xxx` isn't reserved *for any use*, but it is reserved *for use as identifiers with file scope*, which almost all of these appear to be.
And identifiers of the other kind are present too, like `__LTC_SET_ASN1_IDENTIFIER`, `_CRC32_NEGL`, and `_Safer_Expand_Userkey`.

**Comment by fuchsia36** [comment]
I'm afraid there are still some violations left. For instance, `_s0` is reserved, and so is `_safer_expand_userkey`. Both appear to have file scope.

**Comment by sjaeckel** [comment]
> For instance, `_s0` is reserved, and so is `_safer_expand_userkey`.

why? 

>  `_Axxx` and `__xxx` is reserved, but not `_xxx`

and they're both clearly `_xxx`

**Comment by fuchsia36** [comment]
No, _xxx is reserved at file scope as well, but you can use it in local identifiers.

Valid:
`int fun(void)
{
    int _a;
}`
Invalid:
`int _fun(void)
{
    int a;
}`

**Comment by karel-m** [comment]
We should perhaps switch naming convention for static functions to `s_funcname` instead of `_funcname`

**Comment by sjaeckel** [comment]
> We should perhaps switch naming convention for static functions to `s_funcname` instead of `_funcname`

looks like we have to do that...

**Comment by sjaeckel** [comment]
@fuchsia36 does #534 now solve this issue?

**Comment by sjaeckel** [comment]
@fuchsia36 fine now?
